src/Lib.P2P/Protocols/Ping1.cs

```diff
@@ -0,0 +1,190 @@
+
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        ///   Send echo requests to a peer.
+        /// </summary>
+        /// <param name="peerId">
+        ///   The peer ID to receive the echo requests.
+        /// </param>
+        /// <param name="count">
+        ///   The number of echo requests to send.  Defaults to 10.
+        /// </param>
+        /// <param name="cancel">
+        ///   Is used to stop the task.  When cancelled, the <see cref="TaskCanceledException"/> is raised.
+        /// </param>
+        /// <returns>
+        ///   A task that represents the asynchronous operation. The task's value is
+        ///   the sequence of <see cref="PingResult"/>.
+        /// </returns>
+        public async Task<IEnumerable<PingResult>> PingAsync(MultiHash peerId,
+            int count = 10,
+            CancellationToken cancel = default(CancellationToken))
+        {
+            var peer = new Peer {Id = peerId};
+            return await PingAsync(peer, count, cancel).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        ///   Send echo requests to a peer.
+        /// </summary>
+        /// <param name="address">
+        ///   The address of a peer to receive the echo requests.
+        /// </param>
+        /// <param name="count">
+        ///   The number of echo requests to send.  Defaults to 10.
+        /// </param>
+        /// <param name="cancel">
+        ///   Is used to stop the task.  When cancelled, the <see cref="TaskCanceledException"/> is raised.
+        /// </param>
+        /// <returns>
+        ///   A task that represents the asynchronous operation. The task's value is
+        ///   the sequence of <see cref="PingResult"/>.
+        /// </returns>
+        public async Task<IEnumerable<PingResult>> PingAsync(MultiAddress address,
+            int count = 10,
+            CancellationToken cancel = default(CancellationToken))
+        {
+            var peer = SwarmService.RegisterPeerAddress(address);
+            return await PingAsync(peer, count, cancel).ConfigureAwait(false);
+        }
+
+        private async Task<IEnumerable<PingResult>> PingAsync(Peer peer, int count, CancellationToken cancel)
+        {
+            var ping = new byte[PingSize];
+            var rng = new Random();
+            var results = new List<PingResult>
+            {
+                new PingResult {Success = true, Text = $"PING {peer}."}
+            };
+            var totalTime = TimeSpan.Zero;
+
+            using (var stream = await SwarmService.DialAsync(peer, ToString(), cancel))
+            {
+                for (var i = 0; i < count; ++i)
+                {
+                    rng.NextBytes(ping);
+
+                    var start = DateTime.Now;
+                    try
+                    {
+                        await stream.WriteAsync(ping, 0, ping.Length).ConfigureAwait(false);
+                        ;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty statement.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)